src/Google/Ads/GoogleAds/Util/FieldMasks.php

```diff
@@ -247,4 +255,15 @@ private static function isWrapperType(Descriptor $descriptor)
     {
         return in_array($descriptor->getClass(), self::$WRAPPER_TYPES);
     }
+
+    // TODO: We can remove this function when it's supported in google/gax-php:
+    // https://github.com/googleapis/gax-php/issues/285
+    /**
+     * @param string $name
+     * @return string the name of hasser function
+     */
+    private static function getHasser(string $name)
+    {
+        return 'has' . ucfirst(Serializer::toCamelCase($name));
```

Comment:
Is this going to cover all cases? I remember there were some issues around PHP and method names, especially those that included a number.

Reply:
I'm not sure what cases you mean?
Anyway, this is the same as the methods implemented by gax-php (the only diff is that "has" is used instead of "get"), so if there are bugs in gax-php's `toCamelCase()`, we should prioritize fixing them in there.

But so far, after many releases, I've not found any bugs in that `toCamelCase()` method, so I'm convinced that this should be fine.

Reply:
I was thinking of fields like `metrics.video_quartile_100_rate`, which we renamed to `metrics.video_quartile_p100_rate` (added `p` before `100`) for v5 to avoid ambiguity when converting to/from camel case and snake case.
Sounds like this isn't a concern, so I'll mark it resolved.